plugins/dicom_viewer/server/__init__.py

```diff
@@ -98,21 +156,36 @@ def parse_file(f):
                 value = coerce(value.value)
                 if value is not None:
                     data[key] = value
-    except Exception:
-        pass  # if an error occurs, probably not a dicom file
+    except dicom.errors.InvalidDicomError:
+        # if this error occurs, probably not a dicom file
+        return None
     return data
 
 
-def process_file(f):
-    f['dicom'] = parse_file(f)
-    return File().save(f)
-
-
 def handler(event):
-    process_file(event.info['file'])
+    """Whenever an additional file is uploaded to a "DICOM item", remove any
+    DICOM metadata that is no longer common to all DICOM files in the item.
+    """
+    file = event.info['file']
+    fileMetadata = parse_file(file)
+    if fileMetadata is None:
+        return
+    item = Item().load(file['itemId'], force=True)
+    if 'dicom' in item:
+        item['dicom']['meta'] = removeUniqueMetadata(item['dicom']['meta'], fileMetadata)
+    else:
+        # In this case the uploaded file is the first of the item
+        item['dicom'] = {}
+        item['dicom']['meta'] = fileMetadata
+        item['dicom']['files'] = []
+    item['dicom']['files'].append(_extractFileData(file, fileMetadata))
```

Comment:
This new file needs to be added in a sorted position within the existing `files` list. A test should verify that it's doing so (e.g. by uploading files one at a time, out of order).

Reply:
If they're uploaded in the reverse order, and the test check if they are well sorted, do you think it's sufficient?

Reply:
The sort here should use `_getDicomFileSortKey`, which will produce the same output every time. It shouldn't matter what order a user uploads single files, the resulting `files` list should always be the same.

Reply:
Yes, this and the test are updated in the two last commit!